zkevm-circuits/src/mpt_circuit.rs

```diff
@@ -246,17 +239,20 @@ impl<F: Field> MPTConfig<F> {
             .try_into()
             .unwrap();
 
-        let memory_columns = (0..5).map(|_| meta.advice_column()).collect::<Vec<_>>();
-
         let mut state_machine = StateMachineConfig::construct(meta);
         let mut rlp_item = MainRLPGadget::default();
-
-        let mut memory = Memory::new(memory_columns);
-        memory.allocate(meta, key_memory(false), MptCellType::MemKeyCTable);
-        memory.allocate(meta, key_memory(true), MptCellType::MemKeySTable);
-        memory.allocate(meta, parent_memory(false), MptCellType::MemParentCTable);
-        memory.allocate(meta, parent_memory(true), MptCellType::MemParentSTable);
-        memory.allocate(meta, main_memory(), MptCellType::MemMainTable);
+        let memory = Memory::new(
+            meta,
+            vec![
+                (MptCellType::MemKeyC, 3),
+                (MptCellType::MemKeyS, 3),
+                (MptCellType::MemParentC, 3),
+                (MptCellType::MemParentS, 3),
+                (MptCellType::MemMain, 3),
+            ],
```

Comment:
Is there a benefit of using an array here instead of separate objects? Not it seems all memory banks need to be created in a single place, otherwise it's possible to just create them where necessary so a bit more flexible (but perhaps not that useful in practice). But I guess mostly just wondering why you changed this.

Reply:
Because the memory kinda works like a trimmed down CM now :) with `query_read` and `query_write`, so intuitively I adopted lots of things from CM. Also In the future we might be able specify more configs for each bank, like currently the "3" means `ThirdPhase`, potentially we may do optional degree reduction as well.

Reply:
Each memory system could have its own set or parameters, so could be reason not to unify it like here but keep it as different functions depending on the memory type.

> potentially we may do optional degree reduction as well.

A big reason to reuse the the CB as much as possible, so everything related to constraints/lookups has to go through there! Otherwise we have to think about that in multiple places instead of just the CB being able to handle all that.